### DISCUSSION TITLE
html-data-attributes

### DISCUSSION THREAD

**Discussion by giscus** [bloggrammer/DevBlog#11] [Announcements]
# html-data-attributes

Discover the power of HTML5 data attributes and JavaScript in this ultimate guide to using custom element data for improved functionality, performance, and dynamic web development. Learn how to create interactive web pages, enhance user experience, and employ advanced techniques for customizing your web elements in modern web development.

https://bloggrammer.com/html-data-attributes

**Comment by LatonCypher** [comment]
This is interesting. you talk about HTML5 and HTML as if they are different. I thought HTML5 is a version of HTML?

**Reply by LatonCypher** [reply]
Does a div element have name. if yes. Then i think, another version of the code might achieve the same thing. 

const userElement = document.querySelector('.user');
userElement.addEventListener('click', function() {
  const userId = this.getAttribute('data-id');
  console.log('Clicked on ' + userElement.name)});

**Reply by bloggrammer** [reply]
When I referred to HTML5, I was specifically mentioning the latest version of HTML that includes modern features, improvements, and new elements compared to older versions of HTML. HTML5 introduced new tags, attributes, and functionalities that enhance web development, multimedia integration, and overall user experience.

So, in essence, HTML5 is a continuation of the HTML standard, bringing with it new capabilities and features. When talking about HTML, it generally refers to all versions of HTML, but when we mention HTML5, we are referring to the specific set of enhancements and features introduced in the fifth version of the language.

**Reply by bloggrammer** [reply]
In HTML, the <div> element itself does not have a "name" attribute. However, you can assign a "name" attribute to a <div> if you have a specific need for it, but it wouldn't have any special significance by default.
